### PR TITLE
Replace jquery calls with native methods for setProgress

### PR DESCRIPTION
**Pull request by ilnytskyi**
Using native js in setProgress improves time of page rendering.
It can be noticeable on huge profiles in apps like Magento 2
On demo link it gives about 200ms
https://noisebynorthwest.github.io/php-spx/demo/report.html?key=spx-full-20191229_175636-06d2fe5ee423-3795-233665123

### CODE REVIEW

**Comment by NoiseByNorthwest** [comment]
Thanks, this is always a good thing to remove an unnecessary / easily avoidable use of jquery.
However I'm wondering how it can improve performance of the part of the rendering where setProgress is involved, the initial load in other words.
Furthermore I do not observe any noticable improvement with reports ranging from 150K calls to 4M calls.
See the screenshot below, which shows 6 successive loadings of a 150K calls report, the first 3 being made with your patch and the next 3 with master. As you can see there is no noticable difference regarding the timing of the call-range tree constructions.
image

How and on which part of this page did you have observed a 200ms improvement ?

File: assets/web-ui/report.html [hunk 1]
```diff
@@ -291,7 +291,7 @@ <h2>Initializing...</h2>
                     return new Promise(resolve => resolve());
                 })
                 .then(() => initDialog.container.fadeOut(500))
-                .catch(e => initDialog.progressInfo.text(e))
+                .catch(e => initDialog.progressInfo.innerText = (e))
```
Comment (NoiseByNorthwest) [inline assets/web-ui/report.html]:
```suggestion
                .catch(e => initDialog.progressInfo.innerText = e)
```

File: assets/web-ui/report.html [hunk 2]
```diff
@@ -248,7 +248,7 @@ <h2>Initializing...</h2>
                     )
                 ))
                 .then(() => {
-                    initDialog.title.text('Initializing widgets...');
+                    initDialog.title.innerText = ('Initializing widgets...');
```
Comment (NoiseByNorthwest) [inline assets/web-ui/report.html]:
```suggestion
                    initDialog.title.innerText = 'Initializing widgets...';
```

File: assets/web-ui/report.html [hunk 3]
```diff
@@ -85,18 +85,18 @@ <h2>Initializing...</h2>
...
+                title: document.querySelector('#init-report p:nth-of-type(1)'),
+                progressInfo: document.querySelector('#init-report p:nth-of-type(2)'),
+                progress: document.querySelector('#init-report .progress'),
+                progressBar: document.querySelector('#init-report .progress > div'),
                 data: {}
             };
 
             function setProgress(title, current, total) {
-                initDialog.title.html(title);
+                initDialog.title.innerHTML = (title);
```
Comment (NoiseByNorthwest) [inline assets/web-ui/report.html]:
```suggestion
                initDialog.title.innerHTML = title;
```

**Comment by ilnytskyi** [comment]
I checked it in chrome profile. I noticed that setProgress uses jquery methods to update the progress bar.
Selection_067

after
Selection_068

**Comment by NoiseByNorthwest** [comment]
Ok, it is not that significant as the gain is 1.5% (2.5% on my computer) of the profile report loading but this is still welcome to decrease the dependency on jquery.

**Comment by NoiseByNorthwest** [comment]
@ilnytskyi thanks !

**Pull request merged**